src/test/test-copy.c

```diff
@@ -50,6 +51,123 @@ TEST(copy_file) {
         unlink(fn_copy);
 }
 
+TEST(copy_tree_replace_file) {
+        _cleanup_free_ char *buf = NULL;
+        char *src, *dest;
+        size_t sz;
```

Comment:
`streq()` will guarantee both strings are the same size so we can drop the size checks